### PR TITLE
remove unnecessary typed_data imports

### PR DESCRIPTION
**Pull request by dimabran**
Removed all the unnecessary commented out typed_data import ignore

Fixes [Issue #128140](https://github.com/flutter/flutter/issues/128140)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for the contribution!

**Review by tarrinneal — APPROVED** [review]
lgtm once @stuartmorgan's requests are addressed.

File: packages/camera/camera_android/test/type_conversion_test.dart [hunk 1]
```diff
@@ -2,10 +2,6 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// TODO(a14n): remove this import once Flutter 3.1 or later reaches stable (including flutter/flutter#104231)
-// ignore: unnecessary_import
```
[the remark this answers is not in the document]
Comment (dimabran) [reply packages/camera/camera_android/test/type_conversion_test.dart:6]: Please note that i also kept this import in the following files:
[packages/camera/camera_android/lib/src/type_conversion.dart](https://github.com/dimabran/packages/blob/remove_unnecessary_imports/packages/camera/camera_android/lib/src/type_conversion.dart)
[packages/camera/camera_avfoundation/lib/src/type_conversion.dart](https://github.com/dimabran/packages/blob/remove_unnecessary_imports/packages/camera/camera_avfoundation/lib/src/type_conversion.dart)
[packages/camera/camera_avfoundation/test/type_conversion_test.dart](https://github.com/dimabran/packages/blob/remove_unnecessary_imports/packages/camera/camera_avfoundation/test/type_conversion_test.dart)
For the same reason..

File: site-shared
Comment (stuartmorgan) [inline site-shared:1]: Why does this PR include an update of the `site-shared` submodule?

**Comment by stuartmorgan** [comment]
Overriding version and changelog checks; `import` changes are dev-only.

**Comment by stuartmorgan** [comment]
For some reason the Cirrus checks didn't start for the last commit, but since everything but the version check was green in the previous commit, and the last commit only reverted the submodule, it's fine to land.

**Comment by stuartmorgan** [comment]
Hm, @tarrinneal could you re-approve? Your approval was cleared somehow.

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**